apps/frontend/src/lib/utils.ts

```diff
@@ -4,3 +4,7 @@ import { twMerge } from 'tailwind-merge';
 export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs));
 }
+export const getRoleFromJwt = (jwt?: string): string => {
+  const dataFromjwt = jwt ? JSON.parse(atob(jwt.split('.')[1])) : null;
+  return dataFromjwt ? dataFromjwt.role : 'UNAUTHORIZED';
+};
```

Comment:
lehet jobb lenne ezt használni, ezzel tudjuk validálni is, ha lejárt akkor is exceptiont dob. Secret-et semmikép se adjuk meg neki (frontend kódba az nem kerülhet be, ki tudnák bányászni a userek), viszont public keyt nem tudom hogyan lehet csinálni jwt secrethez, de szerintem egy utánanézést megér
https://www.npmjs.com/package/jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback

Reply:
De amúgy a middlewarek nem szerveroldalon futnak le nextben?

Reply:
Lehet igazad van. Kipróbálhatod olyan env varral, aminek nincs next_public az elején, ha az elérhető akkor elvileg szerveren van xd